lib/ssh_scan/api.rb

```diff
@@ -158,12 +180,21 @@ class API < Sinatra::Base
       post '/work/results/:worker_id/:uuid' do
         worker_id = params['worker_id']
         uuid = params['uuid']
+        result = JSON.parse(request.body.first).first
+        socket = {}
+        socket[:target] = result['ip']
+        socket[:port] = result['port']
 
         if worker_id.empty? || uuid.empty?
           return {"accepted" => "false"}.to_json
         end
 
-        settings.db.add_scan(worker_id, uuid, JSON.parse(request.body.first).first)
+        available_result = settings.db.fetch_available_result(socket)
+        unless available_result.nil?
+          settings.db.delete_scan(available_result[:uuid]) unless
```

Comment:
What's the reasoning behind deleting the scan from the database?

I would ideally like to keep this data around so I can query it for research/debugging/performance purposes.